### PR TITLE
fix: Remove excess margin from the top of the canvas

### LINKED ISSUE #20750: [Bug]: UI Polish: Reduce margin at the top of the canvas
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

We're using up a lot of vertical screen real-estate by having a large margin at the top.
We expect more real-estate to be used for what matters -- canvas UI building.

Screenshot 2023-02-17 at 6 42 10 PM

### Steps To Reproduce

Open the editor on the cloud version of Appsmith and notice the margin above the canvas and below the header.

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

All versions

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
This is UI polish fix. 
We're reducing the margin at the top of the canvas to gain vertical real-estate

Fixes #20750 

Media
Before:
Screenshot 2023-02-17 at 6 42 10 PM

After
Screenshot 2023-02-17 at 6 44 25 PM

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by akash-codemonk — DISMISSED** [review]

**Comment by rahulramesha** [comment]
@riodeuno Vercel build seems to have failed

File: app/client/src/pages/Editor/WidgetsEditor/CanvasContainer.tsx [hunk 1]
```diff
@@ -107,8 +107,7 @@ function CanvasContainer() {
       }
       className={classNames({
         [`${getCanvasClassName()} scrollbar-thin`]: true,
-        "mt-0": !shouldHaveTopMargin,
```
Comment (akash-codemonk) [inline app/client/src/pages/Editor/WidgetsEditor/CanvasContainer.tsx:110]: Single page in preview mode has a top margin with this change.
PR
pr
Release
release

Comment (riodeuno) [reply app/client/src/pages/Editor/WidgetsEditor/CanvasContainer.tsx:110]: Thanks @akash-codemonk I'll check it out.

Comment (riodeuno) [reply app/client/src/pages/Editor/WidgetsEditor/CanvasContainer.tsx:110]: @akash-codemonk I've updated the code. Please review again.

**Review by akash-codemonk — DISMISSED** [review]

**Review by rahulramesha — DISMISSED** [review]

**Comment by riodeuno** [comment]
@dhruvikn I would need some context on how the margins are changing.

Context on this PR:
- We're trying to reduce the gap at the top in the "editor" (not the view mode or preview mode)
- The way I've done this is to utilise the `classNames` in `CanvasContainer`, and add smaller margins by checking if we're in `isPreviewMode`. So, if `isPreviewMode = false`, we use smaller margins (tailwind CSS classes)

Please review this. If you have any comments on this, let's discuss it on this thread.

**Comment by dhruvikn** [comment]
Hi @riodeuno,

There are some flags that decide how much the top margin the canvas should have. Here's a breakdown - 

1. isPreviewMode - True when the preview mode (triggered from the top right preview button)
2. isAppSettingsPaneWithNavigationTabOpen - True when the app settings pane is open and the navigation sub-section is active. The canvas here should look exactly as in the preview mode (except for the width).
3. shouldHaveTopMargin - this flag is calculated by
```js
shouldHaveTopMargin = !isPreviewMode || !isAppSettingsPaneWithNavigationTabOpen || pages.length > 1;
```
4. showCanvasTopSection - True when "Start from a template" and "Generate page from a data table" cards are shown on top of the canvas when a user creates a new application.

Using a combination of these flags, we add different tailwind classes to the #canvas-viewport.

I checked out your branch but couldn't find any difference in the top margin.

Your branch - 
CleanShot 2023-03-28 at 15 58 46

Release - 
CleanShot 2023-03-28 at 15 59 04

We will need to change line 278, which has the class "mt-8" as this is the margin in the edit mode.

Also, we can improve on this code chunk. Instead of 
```js
      className={classNames({
        [`${getCanvasClassName()} scrollbar-thin`]: true,
        "mt-0": !shouldHaveTopMargin && isPreviewMode,
        "mt-4": showCanvasTopSection && !isPreviewMode,
        "mt-8":
          shouldHaveTopMargin &&
          !showCanvasTopSection &&
          !isPreviewMode &&
          !isAppSettingsPaneWithNavigationTabOpen,
      })}
```
we can do 
```js 
      className={classNames({
        [`${getCanvasClassName()} scrollbar-thin`]: true,
        "mt-0": !shouldHaveTopMargin,
        "mt-4":
          showCanvasTopSection &&
          !isPreviewMode &&
          !isAppSettingsPaneWithNavigationTabOpen,
        "mt-8": shouldHaveTopMargin && !showCanvasTopSection,
      })}
```

I should have made this small improvement but I missed this in my last improved app navigation PR. Therefore, since you're making changes, can you please do the needful?

**Review by dhruvikn — CHANGES_REQUESTED** [review]
Requesting changes as mentioned here - https://github.com/appsmithorg/appsmith/pull/20751#issuecomment-1486626646

**Comment by riodeuno** [comment]
@dhruvikn Thanks for the feedback. I'm going to be closing this PR in favour of the changes coming in with Auto Layout.

**Comment by riodeuno** [comment]
@prsidhu Here is a cleaner implementation that we were considering. Please see the sample provided by @dhruvikn and consider them in the implementation that we have for auto layout.

**Comment by prsidhu** [comment]
@riodeuno @dhruvikn We'll include the changes derived from this conversation in our next release.

**Pull request closed without merging**